core/src/operations/mod.rs

```diff
@@ -387,6 +387,13 @@ impl Operation {
     pub const OP_BITS: usize = 7;
 
     /// Returns the opcode of this operation.
+    ///
+    /// Opcode patterns have the following meanings:
+    /// - 1001xxx: a u32 or field operation that results in a left shift of the stack, but does not
+    ///   require range checks.
```

Comment:
Are there any u32 operations which would fall under this prefix? If not, should we change this to just "1001xxx: operation that results in a left shift of the stack, but does not require range checks."?

Reply:
This prefix includes `u32and`, `u32or`, `u32not` and the 5 field operations that cause left shifts (`add`, `mul`, `and`, `or`, `eq`). I did this to finish setting the u32 opcodes as requested and group them with more logically similar operations. I left the io, stack, and sys ops to be grouped together. As noted in the comment at the top, those could be given the prefix 0001, and then we could change the comment to be something like "x001xxx: operation that results..." as you suggested. I made the comment more specific for now because the other 7 opcodes have not yet been set.

Reply:
I updated it a bit according to the discussion in #203. See what you think